### PR TITLE
Use rest mail store

### PR DESCRIPTION
**Pull request by Zanthelo**

### CODE REVIEW

File: src/smtp-lib/mail_store.cpp [hunk 1]
```diff
@@ -1,51 +1,55 @@
...
+    json["received_from"] = "sending.server";
+    json["received_by"] = "receiving.server";
+    json["mail_from"] = mail.from;
+    json["rcpt_to"] = mail.to;
+    json["data"] = mail.data;
+
+    for (auto i{0}; i < 5; ++i) {
+        auto response =
+            cpr::Post(cpr::Url{"http://localhost:5000/mailstore"}, cpr::Header{{"Content-Type", "application/json"}},
+                      cpr::Body{json.dump(4)}, cpr::Timeout{5000});
```
Comment (lawben) [inline src/smtp-lib/mail_store.cpp]: what does the `dump(4)` do?

**Review by lawben — DISMISSED** [review]
Can't test this right now because github is stupid. Looks good :)

File: src/smtp-lib/mail_store.cpp [hunk 2]
```diff
@@ -1,51 +1,55 @@
...
+
+        if (response.error.code == cpr::ErrorCode::OK) {
+            break;
```
Comment (lawben) [inline src/smtp-lib/mail_store.cpp]: return true; here

File: src/smtp-lib/mail_store.cpp [hunk 3]
```diff
@@ -1,51 +1,55 @@
...
+        }
```
Comment (lawben) [inline src/smtp-lib/mail_store.cpp]: return false; here

**Review by lawben — CHANGES_REQUESTED** [review]

File: src/smtp-lib/mail_store.cpp (hunk 1 above)
Comment (Zanthelo) [reply src/smtp-lib/mail_store.cpp]: That's the indent used for formatting the string

**Review by lawben — APPROVED** [review]

**Pull request merged**